src/Http/Routing/src/ValidationEndpointFilterFactory.cs

```diff
@@ -25,12 +28,10 @@ public static EndpointFilterDelegate Create(EndpointFilterFactoryContext context
 
         var serviceProviderIsService = context.ApplicationServices.GetService<IServiceProviderIsService>();
 
-        var parameterCount = parameters.Length;
-        var validatableParameters = new IValidatableInfo[parameterCount];
-        var parameterDisplayNames = new string[parameterCount];
-        var hasValidatableParameters = false;
+        // Use a list to only store validatable parameters instead of arrays for all parameters
+        var validatableParameters = new List<ValidatableParameterEntry>();
```

Comment:
Could avoid this allocation if you wanted for endpoints without any validatable params by using `?= new()` when adding the first item.